mm2src/coins/lightning/ln_utils.rs

```diff
@@ -202,42 +277,93 @@ pub async fn start_lightning(ctx: &MmArc, conf: LightningConf) -> EnableLightnin
         }
     }
 
-    // Initialize the ChannelManager to starting a new node without history
-    // TODO: Add the case of restarting a node
     let mut user_config = UserConfig::default();
-
     // When set to false an incoming channel doesn't have to match our announced channel preference which allows public channels
-    // TODO: Add user config to LightningConf maybe get it from coin config
+    // TODO: Add user config to LightningCoinConf maybe get it from coin config / also add to lightning context
     user_config
         .peer_channel_config_limits
         .force_announced_channel_preference = false;
 
-    let best_block = conf
-        .rpc_client
-        .get_best_block()
-        .compat()
-        .await
-        .mm_err(|e| EnableLightningError::RpcError(e.to_string()))?;
-    let best_block_hash =
-        sha256d::Hash::from_slice(&best_block.hash.0).map_to_mm(|e| EnableLightningError::HashError(e.to_string()))?;
-    let chain_params = ChainParameters {
-        network: conf.network,
-        best_block: BestBlock::new(BlockHash::from_hash(best_block_hash), best_block.height as u32),
+    let mut restarting_node = true;
+    let rpc_client = match &filter.clone().unwrap().platform_coin.as_ref().rpc_client {
```

Comment:
Is unwrap safe here? Please add a comment if yes or return the error otherwise.

Reply:
> Is unwrap safe here? Please add a comment if yes or return the error otherwise.

filter which implements the Filter trait https://github.com/KomodoPlatform/atomicDEX-API/blob/7ff4b28217a2e12a1e9369ae883b7f2e145e9071/mm2src/coins/lightning/ln_rpc.rs#L138
is only needed for electrum clients that's why I chose to make it Some here https://github.com/KomodoPlatform/atomicDEX-API/blob/7ff4b28217a2e12a1e9369ae883b7f2e145e9071/mm2src/coins/lightning/ln_utils.rs#L247 for now until Native client is implemented then this code will be changed depending on which client is used.

> What if rpc_client is initialized before platform_coin is moved?

I chose to use filter in places where the code will be used for electrum clients only in the future as a reminder. Maybe I should add a comment in the code as a reminder also.